pkg/check/gc/reserve.go

```diff
@@ -0,0 +1,137 @@
+package gc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"math/rand"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/beeclient/api"
+	"github.com/ethersphere/beekeeper/pkg/random"
+)
+
+func CheckReserve(c *bee.Cluster, o Options) error {
+	ctx := context.Background()
+	rnd := random.PseudoGenerator(o.Seed)
+	fmt.Println("gcc: reserve check")
+	fmt.Printf("Seed: %d\n", o.Seed)
+	ng := c.NodeGroup(o.NodeGroup)
+	pivotNode := ng.NodesSorted()[rnd.Intn(ng.Size())]
+	client := ng.NodeClient(pivotNode)
+	fmt.Printf("node %s\n", pivotNode)
+
+	adrs, err := client.Addresses(ctx)
+	if err != nil {
+		return err
+	}
+
+	origState, err := client.ReserveState(ctx)
+	if err != nil {
+		return fmt.Errorf("reservestate: %w", err)
+	}
+	fmt.Println("reservestate:", origState)
+
+	depth := capacityToDepth(origState.Radius, origState.Available) + 1
```

Comment:
here we'd like to buy a low value batch but after the node picks it up from chain we should expect that the available capacity > 0. once you buy the next batch, capacity should go to 0 and then bounce back after evicting some/all chunks of this batch